File: medium/1689/main.c
int minPartitions(char* n) {
  int max = 0;

  while (*n != '\0') {
    if (*n == '9') {
      return 9;
    }
    if (*n - '0' > max) {
      max = *n - '0';
    }
    n++;
  }

  return max;
    
}

File: medium/1689/main.rs
impl Solution {
    pub fn min_partitions(n: String) -> i32 {
        let mut max = 0;

        for &b in n.as_bytes() {
            if b'9' == b {
                return 9;
            }
            if b - b'0' > max {
                max = b - b'0';
            }
        }
        
        max as i32
    }
}
